### DISCUSSION TITLE
捐献 libxray 至 Xray 社区

### DISCUSSION THREAD

**Discussion by ghost** [XTLS/Xray-core#2142] [Show and tell]
计划捐献 libxray 给 Xray 社区。   @yuhan6665 @RPRX 

libxray 聚焦于改善 Xray-core 在移动端的开发体验。

注：Xray Json 为 Xray 配置的子集，为出口节点添加了 Name 字段，便于 App 内进行解析。

目前计划以下功能：

1. 启动和停止。(完成)
2. 强制 GC。(在 R 大指导下完成)
3. 获取空闲端口。(完成)
4. 测速。(完成)
5. 转换 Clash yaml，Clash.Meta yaml 为 Xray Json。(完成)
6. 转换 v2rayN 订阅为  Xray Json。(进行中，预计不支持 VMessQRCode)
7. 转换 VMessAEAD/VLESS 分享为  Xray Json。
8. 试验性的 Android Protect(fd)。(完成)
9. 读取 geosite 的类别名称，包含 Attribute；读取 geoip 的类别名称。(完成)
10. 编译脚本。一键生成 xcframework 和 aar。(完成)

该项目虽然基于 Xray-core，但大部分代码均自行编写或参考自 v2fly ，鲜有直接复制 Xray-core 代码并修改的地方。所以希望使用商业友好的 MIT License.

不知是否接收该项目？

**Comment by GreatMichaelLee** [comment]
小白纯点赞

**Comment by RPRX** [comment]
可行

**Reply by ghost** [reply]
> 可行

好的。仓库建好后，会直接发一个 PR 过去。

**Reply by RPRX** [reply]
> > 可行
> 
> 好的。仓库建好后，会直接发一个 PR 过去。

到时我可以拉你进来，并转移仓库

**Reply by ghost** [reply]
> > > 可行
> > 
> > 
> > 好的。仓库建好后，会直接发一个 PR 过去。
> 
> 到时我可以拉你进来，并转移仓库

了解。

**Reply by ghost** [reply]
> 可行

额，这事还继续吗？

**Reply by yuhan6665** [reply]
> > 可行
> 
> 额，这事还继续吗？

已发了邀请

**Comment by ghost** [comment]
> 1. 启动和停止。

https://github.com/yiguous/libxray

**Comment by ghost** [comment]
1.0 完工。

**Reply by ghost** [reply]
已在 FoXray 开启内测。

**Comment by 50infivedays** [comment]
我对这个项目很感兴趣 可以讨论下吗
